### PR TITLE
Update removal_of_types.asciidoc

### PR DESCRIPTION
**Pull request by danochoa**
Fix link to [parent-child relationship]

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: docs/reference/mapping/removal_of_types.asciidoc [hunk 1]
```diff
@@ -41,7 +41,7 @@ field, so documents of different types with the same `_id` could exist in a
 single index.
 
 Mapping types were also used to establish a
-/guide/en/elasticsearch/reference/5.4/mapping-parent-field.html[parent-child relationship]
+https://www.elastic.co/guide/en/elasticsearch/reference/5.4/mapping-parent-field.html[parent-child relationship]
```
Comment (cbuescher) [inline docs/reference/mapping/removal_of_types.asciidoc]: I think it whould be better to use the asciidoc syntax for internal cross-references (<<url,link text>>) so we keep linking the current version etc... The page that the hard link refers to has the section title `mapping-parent-field`, so the link should be <<mapping-parent-field,parent-child relationship>>

**Review by cbuescher — CHANGES_REQUESTED** [review]

**Comment by cbuescher** [comment]
@danochoa thanks for opening this PR. I left a comment on how to improve the linking to other parts of the document. After you changed this, would you also sign the CLA so we can merge the PR?

**Comment by danochoa** [comment]
@cbuescher thanks for the feedback, I made the change and signed the CLA

**Review by cbuescher — APPROVED** [review]
@danochoa thanks a lot, LGTM. I will merge this to the active branches where this is still a static link.

**Pull request merged**